### PR TITLE
HPCC4J-537 Log open / close operation info

### PR DESCRIPTION
**Pull request by jpmcmu**
- Added open / close logging output

Signed-off-by: James McMullan James.McMullan@lexisnexis.com

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [X] I have created a corresponding JIRA ticket for this submission
- [X] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [X] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [X] The change has been fully tested:
  - [ ] This change does not cause any existing JUnits to fail.
  - [ ] I have include JUnit coverage to test this change
  - [ ] I have performed system test and covered possible regressions and side effects.
- [X] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] This change fixes the problem, not just the symptom

### CODE REVIEW

**Comment by jpmcmu** [comment]
@rpastrana please review

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/HPCCRemoteFileWriter.java [hunk 1]
```diff
@@ -85,6 +87,12 @@ public HPCCRemoteFileWriter(DataPartition dp, FieldDef recordDef, IRecordAccesso
 
         this.binaryRecordWriter = new BinaryRecordWriter(this.outputStream);
         this.binaryRecordWriter.initialize(this.recordAccessor);
+
+        log.info("HPCCRemoteFileWriter: Opening file part: " + dataPartition.getThisPart()
+                + " compression: " + fileCompression.name() + " "
+                + " record definition:\n"
+                + RecordDefinitionTranslator.toJsonRecord(this.recordDef));
```
Comment (rpastrana) [inline dfsclient/src/main/java/org/hpccsystems/dfs/client/HPCCRemoteFileWriter.java:94]: What's the bloat factor of this output?
Was there discussion of reserving this output for error/warning situations only?

**Review by rpastrana — COMMENTED** [review]
@jpmcmu looks fine, one question about reporting the rec definition every time. 
Also, this should be accompanied by a wiki entry.

**Review by rpastrana — APPROVED** [review]
@jpmcmu approved. Please add link to wiki entry.

**Comment by jpmcmu** [comment]
@rpastrana Added the logging information under the publish failure entry as we had discussed: https://github.com/hpcc-systems/hpcc4j/wiki/Common-Issues-and-Solutions#publish-failed-due-to-missing-file-parts

**Pull request merged**